earthaccess/store.py

```diff
@@ -200,40 +199,49 @@ def get_s3fs_session(
         Returns:
             a s3fs file instance
         """
-        if self.auth is not None:
-            if not any([concept_id, daac, provider, endpoint]):
-                raise ValueError(
-                    "At least one of the concept_id, daac, provider or endpoint"
-                    "parameters must be specified. "
-                )
+        if self.auth is None:
+            raise ValueError(
+                "A valid Earthdata login instance is required to retrieve S3 credentials"
+            )
+        if not any([concept_id, daac, provider, endpoint]):
+            raise ValueError(
+                "At least one of the concept_id, daac, provider or endpoint"
+                "parameters must be specified. "
+            )
+
+        # Get existing S3 credentials if we already have them
+        location = (concept_id, daac, provider, endpoint)
+        need_new_creds = False
+        try:
+            dt_init, creds = self._s3_credentials[location]
```

Comment:
```suggestion
            # Check for cached credentials
            dt_init, creds = self._s3_credentials[location]
```

Reply:
I think this context is already provided with the 

```python
# Get existing S3 credentials if we already have them
```

comment a few lines above